sdk/src/test/java/com/google/cloud/dataflow/sdk/transforms/DoFnTest.java

```diff
@@ -188,4 +193,30 @@ public void processElement(DoFn<String, String>.ProcessContext c) throws Excepti
 
     return pipeline;
   }
+  
+  @Test
+  public void testPopulateDisplayDataDefaultBehavior() {
+    DoFn<String, String> usesDefault =
+        new DoFn<String, String>() {
+          @Override
+          public void processElement(DoFn<String, String>.ProcessContext c) throws Exception {}
+        };
+
+    DisplayData data = collectDisplayData(usesDefault);
+    assertThat(data.items(), empty());
+  }
+
+  private DisplayData collectDisplayData(final DoFn<String, String> fn) {
```

Comment:
I think the more general `DisplayData.from(HasDisplayData root)` is fine. In general, no one will interact with that, since it is just the method the SDK uses to get display data from a root. And if a user did use it directly, it is really only useful for testing unless they have some other way of attaching the data to something. But, if a runner *wanted* to get the data just from a `DoFn` why should we stop them?